nutils/topology.py

```diff
@@ -308,11 +308,12 @@ def project(self, fun, onto, geometry, ischeme='gauss', degree=None, droptol=1e-
       F = numpy.zeros(onto.shape[0])
       W = numpy.zeros(onto.shape[0])
       I = numpy.zeros(onto.shape[0], dtype=bool)
-      fun = function.asarray(fun).prepare_eval(ndims=self.ndims)
-      data = evaluable.Tuple(evaluable.Tuple([fun, onto_f.simplified, evaluable.Tuple(onto_ind)]) for onto_ind, onto_f in evaluable.blocks(onto.prepare_eval(ndims=self.ndims)))
-      for ref, trans, opp in zip(self.references, self.transforms, self.opposites):
-        ipoints = ref.getpoints('bezier2')
-        for fun_, onto_f_, onto_ind_ in data.eval(_transforms=(trans, opp), _points=ipoints, **arguments or {}):
+      ielem_arg = evaluable.Argument('_project_index', (), dtype=int)
```

Comment:
This (ant subsequent occurrences in the same patch) would be another place where `Argument` intbounds could be useful.